rootfs/etc/nginx/lua/balancer/static_balancers.lua

```diff
@@ -0,0 +1,83 @@
+local util = require('util')
+local split = require('util.split')
+local ngx_upstream = require("ngx.upstream")
+local implementations = require('balancer.implementations')
+
+local static_backends = {}
+local static_balancers = {}
+
+local DEFAULT_LB_ALG = "ewma"
+
+local _M = {}
+
+local function marshal_endpoint(endpoint)
+    if (not endpoint.address) or (not endpoint.port) then
+        if endpoint.addr then
+            local addr, err = split.parse_addr(endpoint.addr)
+            if err then
+                return nil, err
+            end
+
+            endpoint.address = addr.host
+            endpoint.port = addr.port
+            endpoint.addr = nil
+
+            return endpoint, nil
+        end
+    end
+    return nil, "error in grabbing address & port" 
+end
+
+local function create_static_backend(upstream_name)
+    local sb = {}
+    sb.name = upstream_name
+
+    sb.endpoints = ngx_upstream.get_servers(upstream_name)
+
+    for index, endpoint in ipairs(sb.endpoints) do
+        sb.endpoints[index] = marshal_endpoint(endpoint)
```

Comment:
I'd avoid updating the variable that you're iterating over. With the changes in my above comment you can do
```
local servers = ngx_upstream.get_servers(upstream_name)
for _, server in ipairs(servers) do
  table.insert(sb.endpoints, marshal_endpoint(server))
end
```